phpcs.xml.dist

```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0"?>
+<ruleset
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="vendor/squizlabs/php_codesniffer/phpcs.xsd"
+>
+    <arg name="basepath" value="."/>
+    <arg name="extensions" value="php"/>
+    <arg name="parallel" value="80"/>
+    <arg name="cache" value=".phpcs-cache"/>
+    <arg name="colors"/>
+    <arg value="nps"/>
+
+    <file>src/</file>
+
+    <rule ref="Doctrine">
+        <exclude name="Generic.Formatting.SpaceAfterNot"/>
+        <exclude name="SlevomatCodingStandard.Classes.SuperfluousAbstractClassNaming"/>
+        <exclude name="SlevomatCodingStandard.Classes.SuperfluousInterfaceNaming"/>
+        <exclude name="SlevomatCodingStandard.Classes.SuperfluousExceptionNaming"/>
+
+        <exclude name="SlevomatCodingStandard.ControlStructures.DisallowYodaComparison.DisallowedYodaComparison"/>
+        <exclude name="SlevomatCodingStandard.ControlStructures.AssignmentInCondition.AssignmentInCondition"/>
+
+        <exclude name="SlevomatCodingStandard.Commenting.RequireOneLinePropertyDocComment.MultiLinePropertyComment"/>
+
+        <exclude name="Generic.Formatting.MultipleStatementAlignment.NotSame"/>
+        <exclude name="Generic.Formatting.MultipleStatementAlignment"/>
+
+        <exclude name="Squiz.Arrays.ArrayDeclaration.KeySpecified"/>
+
+        <exclude name="SlevomatCodingStandard.ControlStructures.EarlyExit"/>
+        <exclude name="SlevomatCodingStandard.TypeHints.NullableTypeForNullDefaultValue"/>
```

Comment:
This one **should** be kept enabled and enforced.

Non-nullable types without default value null are a legacy behavior that should not be relied upon.
See discussion here for reasoning: https://github.com/doctrine/common/pull/805#discussion_r128721368

Also to quote [Typed Properties RFC](https://wiki.php.net/rfc/typed_properties_v2) (targeting 7.3):
> Typed properties **cannot have a null default value, unless the type is explicitly nullable (?Type)**. This is in contrast to parameter types, where a null default value automatically implies a nullable type. **We consider this to be a legacy behavior**, which we do not wish to support for newly introduced syntax.